dashboard/modules/job/common.py

```diff
@@ -156,7 +159,17 @@ def get_all_jobs(self) -> Dict[str, JobInfo]:
                 self.JOB_DATA_KEY_PREFIX
             ), "Unexpected format for internal_kv key for Job submission"
             job_ids.append(job_id_with_prefix[len(self.JOB_DATA_KEY_PREFIX) :])
-        return {job_id: self.get_info(job_id) for job_id in job_ids}
+
+        async def get_job_info(job_id: str):
+            job_info = await self.get_info(job_id)
+            return job_id, job_info
+
+        return {
+            job_id: job_info
+            for job_id, job_info in await asyncio.gather(
+                *[get_job_info(job_id) for job_id in job_ids]
```

Comment:
Btw, this is not a blocker for the PR. But I wonder @iycheng do we have APIs to get a list of entities from the internal kv?